ext/afform/core/Civi/Api4/Action/Afform/AbstractProcessor.php

```diff
@@ -272,30 +272,44 @@ protected static function getJoinWhereClause(FormDataModel $formDataModel, strin
     $mainEntityType = $entity['type'];
     $params = [];
 
+    // Forward FK e.g. Event.loc_block_id => LocBlock
+    $forwardFkField = self::getFkField($mainEntityType, $joinEntityType);
+    if ($forwardFkField) {
+      $joinIdField = $forwardFkField['fk_column'];
+      $mainEntityJoinValue = civicrm_api4($mainEntityType, 'get', [
+        'checkPermissions' => FALSE,
+        'where' => [['id', '=', $mainEntityId]],
+        'select' => [$forwardFkField['name']],
+      ])->first();
+      return [$joinIdField, '=', $mainEntityJoinValue[$forwardFkField['name']] ?? 0];
```

Comment:
@colemanw Should this return an array of current return value? I get the following error when trying to prefill the afform with event id in URL (happens only if location block is included).

>TypeError: array_pad(): Argument #1 ($array) must be of type array, string given in array_pad() (line 303 of web/sites/all/modules/civicrm/Civi/Api4/Query/Api4SelectQuery.php).